router/mount.go

```diff
@@ -11,18 +13,43 @@ var HttpMethods = []string{"OPTIONS", "GET", "HEAD", "POST", "PUT", "DELETE", "P
 // ServerlessRouterMount represents a single mount point on the API
 // Such as '/path', the HTTP method, and the function to resolve it
 type ServerlessRouterMount struct {
-	Name     		string
-	Function 		*AWSServerlessFunction
-	Handler  		http.HandlerFunc
-	Path     		string
-	Method   		string
+	Name             string
+	Function         *AWSServerlessFunction
+	Handler          http.HandlerFunc
+	Path             string
+	Method           string
+	BinaryMediaTypes []string
 
 	// authorization settings
 	AuthType       string
 	AuthFunction   *AWSServerlessFunction
 	IntegrationArn *LambdaFunctionArn
 }
 
+// Returns the wrapped handler to encode the body as base64 when binary
+// media types contains Content-Type
+func (m *ServerlessRouterMount) WrappedHandler() http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		contentType := req.Header.Get("Content-Type")
+		binaryContent := false
+
+		for _, value := range m.BinaryMediaTypes {
+			if value == contentType {
+				binaryContent = true
+				break
+			}
+		}
+
+		if binaryContent {
+			if body, err := ioutil.ReadAll(req.Body); err == nil {
+				req.Body = ioutil.NopCloser(strings.NewReader(base64.StdEncoding.EncodeToString(body)))
+			}
```

Comment:
The proxy request object also includes an `isBase64Encoded` property - sample event [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-input-format). We should add it to the `Event` struct with `false` as a default value and set it to `true` here.

Reply:
Thanks! Good to know it!
Unfortunately, I needed to change HandlerFunc interface to pass isBase64Encoded to the place where NewEvent is created.
I couldn't come up with a good way without changing it.
Please let me know if there's a better way to do this.